pkg/server/commands/reverseexpand/reverse_expand_weighted.go

```diff
@@ -491,3 +529,232 @@ func (c *ReverseExpandQuery) buildFilteredIterator(
+					Consistency:          req.Consistency,
+					RequestMetadata:      graph.NewCheckRequestMetadata(),
+				}, userset)
+			tmpCheckResult, err := handlerFunc(ctx)
+			if err != nil {
+				functionName := "intersectionHandler"
+				if !isAllowed {
+					functionName = "exclusionHandler"
+				}
+				c.logger.Error("Failed to execute", zap.Error(err),
+					zap.String("function", functionName),
+					zap.String("object", tmpResult.Object),
+					zap.String("relation", req.Relation),
+					zap.String("user", req.User.String()))
+				return err
+			}
+
+			// Check if the check result matches the expect isAllowed value
+			// eg, for intersection we expect the check result to be true
+			// and for exclusion we expect the check result to be false.
+			if tmpCheckResult.GetAllowed() == isAllowed {
+				if stack.Len(req.relationStack) == 0 {
+					// If the original stack only had 1 value, we can trySendCandidate right away (nothing more to check)
+					c.trySendCandidate(ctx, false, tmpResult.Object, resultChan)
+				} else {
+					// If the original stack had more than 1 value, we need to query the parent values
+					// new stack with top item in stack
+					err = c.queryForTuples(ctx, req, false, resultChan, tmpResult.Object)
+					if err != nil {
+						return err
+					}
+				}
+			}
+			// otherwise, candidates are not true candidate and no need to pass back to its parents.
+		}
+		return nil
+	})
+}
+
+// invoke loopOverWeightedEdges to get list objects candidate. Check
+// will then be invoked on the non-lowest weight edges against these
+// list objects candidates. If check returns true, then the list
+// object candidates are true candidates and will be returned via
+// resultChan. If check returns false, then these list object candidates
+// are invalid because it does not satisfy all paths for intersection.
+func (c *ReverseExpandQuery) intersectionHandler(
+	ctx context.Context,
+	req *ReverseExpandRequest,
+	resultChan chan<- *ReverseExpandResult,
+	edges []*weightedGraph.WeightedAuthorizationModelEdge,
+	sourceUserType string,
+	resolutionMetadata *ResolutionMetadata,
+) error {
+	intersectionEdgeComparison, err := typesystem.GetEdgesForIntersection(edges, sourceUserType)
+	if err != nil {
+		c.logger.Error("Failed to get lowest weight edge",
+			zap.String("function", "intersectionHandler"),
+			zap.Error(err),
+			zap.Any("edges", edges),
+			zap.String("sourceUserType", sourceUserType))
+		return err
+	}
+
+	if !intersectionEdgeComparison.DirectEdgesAreLeastWeight && intersectionEdgeComparison.LowestEdge == nil {
+		// no need to go further because list objects must return empty
+		return nil
+	}
+
+	var lowestWeightEdges []*weightedGraph.WeightedAuthorizationModelEdge
+	if intersectionEdgeComparison.DirectEdgesAreLeastWeight {
+		lowestWeightEdges = intersectionEdgeComparison.DirectEdges
+	} else {
+		lowestWeightEdges = []*weightedGraph.WeightedAuthorizationModelEdge{intersectionEdgeComparison.LowestEdge}
```

Comment:
You could set this as default on line 659 and change it on 661 effectively removing the else.